src/client/react.tsx

```diff
@@ -39,10 +39,12 @@ export type BoardProps<G extends any = any> = ClientState<G> &
     isMultiplayer: boolean;
   };
 
+export type CtxClient = Omit<Ctx, 'random' | '_random'>
+
 type ReactClientOpts<
   G extends any = any,
   P extends BoardProps<G> = BoardProps<G>,
-  CtxWithPlugins extends Ctx = Ctx
+  CtxWithPlugins extends CtxClient = CtxClient
```

Comment:
I’m not sure this is the correct place for this. Firstly, [the plain JS client](https://github.com/boardgameio/boardgame.io/blob/main/src/client/client.ts) is the main source of truth for client types — the React client wraps it so they should be in sync. Secondly and slightly counter-intuitively, this type here is not actually used for `ctx` on the client I don’t think — it’s just here for compatibility with games that use custom plugins, so this type actually *should* include plugin APIs! (Confusing, I know.)

Probably `ctx` on the client wants updating here: https://github.com/boardgameio/boardgame.io/blob/b406e5a3ca1efa041de5c2b3d88cff6a773e0a0c/src/client/client.ts#L128-L134

Although, I’ll be interested to see if that’s enough, because internally we add and remove the plugin APIs at various points, so having `random` etc. no longer optional may cause errors there.